controllers/teams.js

```diff
@@ -0,0 +1,201 @@
+        if ( !team ) return res.status( 404 ).send( 'No team with that ID' );
+
+        for ( var i in team.players ) {
+            if ( team.players[ i ].playerId.toString() === req.body.playerId.toString() ) {
+                console.log( 'about to toggle' )
+                team.players[ i ].isReady = !team.players[ i ].isReady;
+                console.log( 'player\'s isReady is toggled' );
+                var status = team.players[ i ].isReady;
+                break;
+            }
+        }
+        team.markModified( 'players' );
+        team.save( ( err ) => {
+            if ( err ) return next( err );
+            console.log( 'player: ' + req.body.playerId + ' updated to: ' + status );
+            return res.sendStatus( 200 );
+        });
+    });
+}
+
+exports.addUserToTeam = ( req, res, next ) => {
+    if ( !req.body.playerId ) return res.status( 400 ).send( 'no player ID' );
+
+    Promise.all([
+        User.findById( req.body.playerId ).exec(),
+        Team.findById( req.body.teamId ).exec()
+
+    ]).then( ( results ) => {
+        let user = results[ 0 ];    // get user
+        let team = results[ 1 ];    // get team
+
+        if ( !team )
+            return res.status( 404 ).send( 'No team with that ID' );
+        if ( team.players.length === 5 ) {
+            var err = new Error( 'Max number of players is 5' );
+            err.status = 400;
+            throw err;
+        }
+
+        // check if player is already member of the team
+        for ( var i in team.players ) {
+            if ( team.players[ i ].playerId == req.body.playerId ) {
+                var err = new Error( 'This player already member of your team' );
+                err.status = 400;
+                throw err;
+            }
+        }
+
+        // add player to the team
+        team.players.push({
+            playerId: req.body.playerId
+        });
+
+        team.markModified( 'players');
+        team.save();
+
+        // add team to player
+        user.team = req.body.teamId;
+        user.markModified( 'team' );
+        user.save();
+
+    }).then( () => {
+        return res.sendStatus( 200 );
+    }).catch( ( err ) => next( err ) );
+}
+
+exports.removeUserFromTeam = ( req, res, next ) => {
+    if ( !req.body.playerId ) return res.status( 400 ).send( 'no player ID' );
+
+    Promise.all([
+        User.findById( req.body.playerId ).exec(),
+        Team.findById( req.body.teamId ).exec()
+
+    ]).then( ( results ) => {
+        let user = results[ 0 ];
+        let team = results[ 1 ];
+
+        let userIndex = team.players.findIndex( player => player.playerId == req.body.playerId );
+        if ( userIndex === -1 ) {
+            var err = new Error( 'No players with that ID' );
+            err.status = 400;
+            throw err;
+        }
+
+        team.players.splice( userIndex, 1 );
+        team.markModified( 'players' );
+        team.save();
+
+        user.team = undefined;

```

Comment:
Users shouldn't have references to teams, it should be the other way around. Because a user might be in multiple team. When you need to get user team, you query the team model not the user model.

ie: its one2many relationship.

Reply:
OKAY